File: src/main/java/org/radarbase/management/config/audit/CustomRevisionListener.java
package org.radarbase.management.config.audit;

import org.hibernate.envers.RevisionListener;
import org.radarbase.auth.config.Constants;
import org.radarbase.management.domain.audit.CustomRevisionEntity;
import org.radarbase.management.domain.support.AutowireHelper;
import org.radarbase.management.security.SpringSecurityAuditorAware;
import org.springframework.beans.factory.annotation.Autowired;

public class CustomRevisionListener implements RevisionListener {

    @Autowired
    private SpringSecurityAuditorAware springSecurityAuditorAware;

    @Override
    public void newRevision(Object revisionEntity) {
        AutowireHelper.autowire(this, springSecurityAuditorAware);
        CustomRevisionEntity entity = (CustomRevisionEntity) revisionEntity;
        entity.setAuditor(springSecurityAuditorAware.getCurrentAuditor()
                .orElse(Constants.SYSTEM_ACCOUNT));
    }
}

File: src/main/java/org/radarbase/management/config/audit/CustomRevisionListener.kt
package org.radarbase.management.config.audit

import org.hibernate.envers.RevisionListener
import org.radarbase.management.domain.audit.CustomRevisionEntity
import org.radarbase.management.domain.support.AutowireHelper
import org.radarbase.management.security.Constants
import org.radarbase.management.security.SpringSecurityAuditorAware
import org.springframework.beans.factory.annotation.Autowired


class CustomRevisionListener : RevisionListener {
    @Autowired
    private val springSecurityAuditorAware: SpringSecurityAuditorAware? = null
    override fun newRevision(revisionEntity: Any) {
        AutowireHelper.autowire(this, springSecurityAuditorAware)
        val entity = revisionEntity as CustomRevisionEntity
        entity.auditor = springSecurityAuditorAware!!.currentAuditor
            .orElse(Constants.SYSTEM_ACCOUNT)
    }
}
